.circleci/config.yml

```diff
@@ -150,49 +154,33 @@ commands:
           command: git submodule update --init
       - emsdk-env
       - npm-install
-      - run:
-          name: run tests
-          command: |
-            ./tests/runner << parameters.test_targets >>
-            $EMSDK_PYTHON ./tools/check_clean.py
-  run-tests-mac:
+      - pip-install
+  run-tests:
     description: "Runs emscripten tests"
     parameters:
       test_targets:
         description: "Test suites to run"
         type: string
     steps:
-      - attach_workspace:
-          # Must be absolute path or relative path from working_directory
-          at: ~/
-      - checkout
-      - emsdk-env
-      - npm-install
-      - run:
-          name: Install brew package dependencies
-          environment:
-            HOMEBREW_NO_AUTO_UPDATE: "1"
-          command: |
-            brew list cmake || brew install cmake
-            brew list python3 || brew install python3
       - run:
           name: run tests
           command: |
             ./tests/runner << parameters.test_targets >>
             $EMSDK_PYTHON ./tools/check_clean.py
+  run-tests-linux:
```

Comment:
why is this called linux? looks like it is very generic, just prepares and then runs tests?

does this overlap with another PR? the `prepare-for-tests` part looks familiar...

Reply:
Yes I have other PR that was trying to do a similar thing on windows: #16378.

But I figured making it work on mac first was a good plan.

Reply:
> But I figured making it work on mac first was a good plan.

Do you mean Linux?

Reply:
Oh, I see. This differentiates the refactored linux steps from the new mac steps.

Reply:
No, we already run all the socket tests on linux.. this change just adds the running on one of them on macOS